### PR TITLE
feat(ui): Add option to sort applications list by name and deploy time

### LINKED ISSUE #4060: Add support for sorting applications by name
# Summary
The Argo UI doesn't currently allow sorting by name or any other attribute such as last sync time.

# Motivation
I would use this when scanning the list of available apps by name, sorting by deployment time, etc.

# Proposal
Client-side sorting of the fields in memory.

### PR DESCRIPTION
**Pull request by rbreeze**
Screen Shot 2022-02-18 at 12 23 19

Fixes #4060

Signed-off-by: Remington Breeze <remington@breeze.software>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [x] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [x] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

**Review by alexmt — CHANGES_REQUESTED** [review]
Thank you! Notificed couple things that can be improved: 
* icons/checkboxes and corresponding text are not on the same line
* checkboxes should be radio buttons instead since user cannot sort but both at the same time

image

What do you think about moving sort options to `Sort by` dropdown next to `Items per page` dropdown?

**Comment by rbreeze** [comment]
I think next to items-per-page is a good location for Sort By options. However, right now the items-per-page dropdown is part of a shared component, so it would require refactoring that component to put a dropdown there. 

I changed the options to radio buttons, and they now change appearance based on screen size. What do you think?

Screen Shot 2021-07-13 at 2 58 01 PM
Screen Shot 2021-07-13 at 3 00 36 PM

**Comment by pasha-codefresh** [comment]
I am not sure that it is usable in terms of UI, maybe better put it on top of filters as you did before just in some better way? 

Снимок экрана 2022-02-20 в 16 29 45

In same style as we have for filters

WHYT @rbreeze

**Comment by dhirschfeld** [comment]
I think this would be a great improvement to the UI (even if not perfect yet).

Is there anything remaining to be addressed before this can be merged?

**Comment by alexef** [comment]
@rbreeze I'm also interested in this, can I help rebase/bring it back to life?

File: ui/src/app/applications/components/applications-list/applications-list.tsx [hunk 1]
```diff
@@ -283,6 +283,45 @@ const FlexTopBar = (props: {toolbar: Toolbar | Observable<Toolbar>}) => {
...
+type AppSorter = (a: models.Application, b: models.Application) => number;
+
+const AppSorters: {[key: string]: AppSorter} = {
+    'Name': ((a, b) => {
+        if (a && a.metadata && b && b.metadata) {
+            return a.metadata.name < b.metadata.name ? -1 : 1;
+        }
+        return 1;
+    }) as AppSorter,
+    'Deployment Time': ((a, b) => (a.status.observedAt < b.status.observedAt ? 1 : -1)) as AppSorter
```
Comment (alexef) [inline ui/src/app/applications/components/applications-list/applications-list.tsx:295]: Looks like the `ApplicationStatus.ObservedAt` field has been deprecated, probably `ApplicationStatus.OperationState.FinishedAt` would be a better replacement?

**Comment by crenshaw-dev** [comment]
@alexef was this superseded by https://github.com/argoproj/argo-cd/pull/11645?

**Comment by alexef** [comment]
@crenshaw-dev yes, and there was a follow up also charging the default order.

I believe this PR can be closed.

**Pull request closed without merging**